extensions/amp-user-notification/0.1/amp-user-notification.js

```diff
@@ -224,6 +227,10 @@ export class AmpUserNotification extends AMP.BaseElement {
   /** @override */
   shouldShow() {
     return this.storagePromise_.then(storage => {
+      if (!this.persistentDismissal_) {
+        // Skip reading storage when not persistent-dismissal.
+        return false;
+      }
```

Comment:
Somewhere, we should also prevent storage from writing when `!persistentDismissal`, right?